source/brailleDisplayDrivers/brailleNote.py

```diff
@@ -260,46 +318,54 @@ def display(self, cells):
+			"kb:leftArrow": ("br(braillenote):space+d3","br(braillenote):wleft","br(braillenote):leftArrow",),
+			"kb:rightArrow": ("br(braillenote):space+d6","br(braillenote):wright","br(braillenote):rightArrow",),
+			"kb:pageup": ("br(braillenote):space+d1+d3","br(braillenote):function+upArrow",),
+			"kb:pagedown": ("br(braillenote):space+d4+d6","br(braillenote):function+downArrow",),
+			"kb:home": ("br(braillenote):space+d1+d2","br(braillenote):function+leftArrow",),
+			"kb:end": ("br(braillenote):space+d4+d5","br(braillenote):function+rightArrow",),
+			"kb:control+home": ("br(braillenote):space+d1+d2+d3","br(braillenote):read+T",),
+			"kb:control+end": ("br(braillenote):space+d4+d5+d6","br(braillenote):read+B",),
+			"kb:enter": ("br(braillenote):space+d8","br(braillenote):wcenter","br(braillenote):enter",),
+			"kb:shift+tab": ("br(braillenote):space+d1+d2+d5+d6","br(braillenote):wcounterclockwise","br(braillenote):shift+tab",),
+			"kb:tab": ("br(braillenote):space+d2+d3+d4+d5","br(braillenote):wclockwise","br(braillenote):tab",),
+			"kb:backspace": ("br(braillenote):space+d7","br(braillenote):backspace",),
+			"showGui": ("br(braillenote):space+d1+d3+d4+d5","br(braillenote):read+N",),
+			"kb:windows": ("br(braillenote):space+d2+d4+d5+d6","br(braillenote):read+W",),
+			"kb:alt": ("br(braillenote):space+d1+d3+d4","br(braillenote):read+M",),
+			"toggleInputHelp": ("br(braillenote):space+d2+d3+d6","br(braillenote):read+1",),
 		},
 	})
 
 class InputGesture(braille.BrailleDisplayGesture, brailleInput.BrailleInputGesture):
 	source = BrailleDisplayDriver.name
 
-	def __init__(self, keys=None, dots=None, space=False, routing=None):
+	def __init__(self, keys=None, dots=None, space=False, routing=None, wheel=None, qtMod=None, qtData=None):
 		super(braille.BrailleDisplayGesture, self).__init__()
-		# see what thumb keys are pressed:
+		# Denotes if we're dealing with a QT model.
+		self.qt = qtMod is not None
+		# Handle thumb-keys and scroll wheel (wheel is for Apex BT).
 		names = set()
 		if keys is not None:
-			names.update(_keyNames[1 << i] for i in xrange(4)
-					if (1 << i) & keys)
+			names.update(_keyNames[1 << i] for i in xrange(4) if (1 << i) & keys)
+		elif wheel is not None:
+			names.add(_scrWheel[wheel])
 		elif dots is not None:
-		# now the dots
 			self.dots = dots
 			if space:
 				self.space = space
 				names.add(_keyNames[0])
-			names.update(_dotNames[1 << i] for i in xrange(8)
-					if (1 << i) & dots)
+			names.update(_dotNames[1 << i] for i in xrange(8) if (1 << i) & dots)
 		elif routing is not None:
 			self.routingIndex = routing
 			names.add('routing')
-		self.id = "+".join(names)
\ No newline at end of file
+		elif qtMod is not None:
+			names.update(_qtKeyNames[1 << i] for i in xrange(4)
+				if (1 << i) & qtMod)
+		# Make sure to display QT identifiers in mod+char format if this is such a case.
+		if self.qt:
+			self.id = qtData if qtMod == 0 else "+".join(("+".join(names), qtData))
+		else:

```

Comment:
Gesture identifiers actually must be in Python set order; you can't reorder things. Otherwise, they might not match gesture maps. If you want to reorder them for display reasons, you will need to override `displayName` and `logIdentifier`. If you want to do that, you'll need a helper method which all of them call which returns the names in the correct order (list not set).